redbot/cogs/streams/main.py

```diff
@@ -41,8 +41,9 @@ def __init__(self, bot: Red):
 
         self.bot = bot
 
-        self.streams = self.bot.loop.create_task(self.load_streams())
-        self.communities = self.bot.loop.create_task(self.load_communities())
+        self.bot.loop.create_task(self.load_streams())
```

Comment:
`self.streams` and `self.communities` must be initialised as an iterable before `_stream_alerts()` is called, otherwise it will raise an `AttributeError` or `type is not iterable` error. This is possible in the current scenario because these three tasks are scheduled to run asynchronously, and not linearly.

You could do this just by initialising them as empty lists before scheduling their `load_X` functions, or otherwise.

Reply:
I thought about an empty pre-initialization too, but I didn't think it'd be needed...

Now that I look at `_stream_alerts`, you might want the 2 previous tasks to complete before that happens though (So the first run of that method is done with a complete list of streams and communities).
So the 2 possible solutions are empty pre-init, or waiting for the tasks. You tell me :)

Reply:
I don't think waiting for tasks is a good idea here, as it will have blocking effects. You could create another async function which awaits the first two tasks before creating the `self._stream_alerts()` task, so you can guarantee linearity. Then you only have to use `create_task` once in the `__init__`